### PR TITLE
Align eventhubs connection property names with Go

### PR DESCRIPTION
**Pull request by LarryOsterman**
Ronnie noticed that C++ Eventhubs isn't being reflected in the user-agent telemetry for eventhubs. Inspecting the Go SDK, there is an important difference between Go and C++. This reflects this change.

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp [hunk 1]
```diff
@@ -265,17 +265,17 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace _detail
     {
       constexpr const char* packageName = "cpp-azure-messaging-eventhubs-cpp";
 
-      options.Properties.emplace("Product", +packageName);
-      options.Properties.emplace("Version", PackageVersion::ToString());
+      options.Properties.emplace("product", +packageName);
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp:268]:
```suggestion
      options.Properties.emplace("product", packageName);
```

File: sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp [hunk 2]
```diff
@@ -265,17 +265,17 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace _detail
     {
       constexpr const char* packageName = "cpp-azure-messaging-eventhubs-cpp";
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp:266]:
```suggestion
      constexpr const char* packageName = "azure-messaging-eventhubs-cpp";
```

**Review by antkmsft — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp (hunk 1 above)
Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp:268]: I'm trying VERY hard to keep this PR to the absolute minimum necessary to fix the problem.

Do you believe that the unary + symbol introduces a problem?

File: sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp [hunk 3]
```diff
@@ -265,7 +265,7 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace _detail
     {
       constexpr const char* packageName = "azure-messaging-eventhubs-cpp";
 
-      options.Properties.emplace("product", +packageName);
+      options.Properties.emplace("product", packageName);
```
Comment (RickWinter) [inline sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp:268]: You say Go, Does this align with all languages?

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp:268]: Go, .Net, and Java. After Java, I stopped looking :).

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp:268]: Also python and js (checked them).

So C++ was the only SDK using upper case.

**Pull request merged**